core/consensus/safety/primitives/src/lib.rs

```diff
@@ -0,0 +1,43 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Common utilities for accountable safety in Substrate.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+/// Trait for equivocation proofs.
+pub trait EquivocationProof<H, S>
+{
+	/// Create an equivocation proof for AuRa or Babe.
+	fn new(
```

Comment:
typically traits don't dictate terms of construction, only interface of usage.
For any header-based equivocation proofs, the header itself contains the signatures.

Other kinds of equivocations:
    - Tendermint: we'll need round #, authority ID, 2 conflicting votes (nothing else needed)
    - GRANDPA: we'll need set,round #, authority ID, 2 conflicting votes

Reply:
I changed the name of the trait to `AuthEquivProof`. The signatures are there because we are removing the seals from the headers and they are not supported afterwards (I think we talked a bit about this).